packages/wrangler/src/publish.ts

```diff
@@ -36,10 +37,163 @@ type Props = {
   dryRun: boolean | undefined;
 };
 
+type Route = Exclude<Config["routes"], undefined>[number];
+type RouteObject<R extends Route = Route> = R extends string ? never : R;
```

Comment:
These type definitions look scary, but they simply extract the object version of a route definition in the config (i.e. `routes: [{ pattern: "api.example.com" }]` rather than `routes: ["api.example.com"]`). A custom domain must be defined in the object form (to provide the flag `custom_domain: true`), so this keeps us in sync whenever the Config definition changes